File: xor/xor.c
#define XOR_KEY 0xFF
#define BUFFER_SIZE 2048

#include <string.h>
#include <stdio.h>
#include <errno.h>
#include <time.h>

int main(void) {
	FILE *inf, *outf;
	char buff[BUFFER_SIZE];
	size_t read_bytes;

	const char *path = "gta3.img";
	char outpath[260] = {0};
	strcpy(outpath, path);
	strcat(outpath, "-xor");

	inf = fopen(path, "rb");
	if (inf == NULL) {
		fprintf(stderr, "Failed to open input file, errno = %d\n", errno);
		return 1;
	}

	outf = fopen(outpath, "wb");
	if (outf == NULL) {
		fprintf(stderr, "Failed to open output file, errno = %d\n", errno);
		return 1;
	}

	while ((read_bytes = fread(buff, 1, sizeof(buff), inf)) > 0) {
		for(int i = 0; i < (int)(sizeof(buff) / sizeof(buff[0])); ++i)
			buff[i] ^= XOR_KEY;
		fwrite(buff, 1, read_bytes, outf);
	}

	fclose(inf);
	fclose(outf);
	return 0;
}

File: xor/xor.php
<?php

$XOR_KEY = 0xFF;
$BUFFER_SIZE = 2048;

$path = 'gta3.img';
$outpath = $path.'-xor';

$inf = fopen($path, "rb") or die("Error");
$outf = fopen($outpath, "wb") or die("Error");

while(!feof($inf)) {
	$buff = fread($inf, $BUFFER_SIZE);
	$r = [];
	$len = strlen($buff);
	for ($i = 1; $i <= $len; $i++) {
		$ch = ord(substr($buff, $i, 1));
		$r[$i] = chr($ch ^ $XOR_KEY);
	}
	fwrite($outf, implode($r), $len);
}

fclose($inf);
fclose($outf);
